### PR TITLE
Consider comparing only URI when checking UFS path

### PR DESCRIPTION
**Pull request by yyongycy**
### What changes were proposed in this pull request?

Fixes Alluxio/new-contributor-tasks#594

### Why are the changes needed?
The authority of default uri could be different with the current one. Just need check the path.

### Does this PR introduce _any_ user-facing change?

No

### CODE REVIEW

**Comment by yyongycy** [comment]
> Automated checks report:
> 
> * Commits associated with Github account: FAIL
>   
>   * It looks like your commits can't be linked to a valid Github account.
>     Your commits are made with the email [yangyong@sensorsdata.cn](mailto:yangyong@sensorsdata.cn), which does not allow your contribution to be tracked by Github.
>     See [this link](https://help.github.com/articles/why-are-my-contributions-not-showing-up-on-my-profile/) for possible reasons this might be happening.
>     To change the author email address that your most recent commit was made under, you can run:
>     ```shell
>     git -c user.name="Name" -c user.email="Email" commit --amend --reset-author
>     ```
>     
>     
>         
>           
>         
>     
>           
>         
>     
>         
>       
>     See [this answer](http://stackoverflow.com/a/1320317/401884) for more details about how to change commit email addresses.
>     Once the author email address has been updated, update the pull request by running:
>     ```shell
>     git push --force https://github.com/yyongycy/alluxio.git new-contributor-tasks-issue594
>     ```
> * PR title follows the conventions: FAIL
>   
>   * The title of the PR does not pass all the checks. Please fix the following issues:
>     
>     * Title is too long (75 characters). Must be at most 72 characters.
>     * First word must be capitalized
>     * First word of title (":") is not an imperative verb. Please use one of the [valid words](https://raw.githubusercontent.com/Alluxio/alluxio/master/docs/resources/pr/pr_title_words.md)
> 
> Some checks failed. Please fix the reported issues and reply 'alluxio-bot, check this please' to re-run checks.

fixed

**Comment by yyongycy** [comment]
> * this answer

trigger verifying

**Comment by yyongycy** [comment]
alluxio-bot, check this please

**Comment by yyongycy** [comment]
alluxio-bot, check this please

**Comment by yyongycy** [comment]
alluxio-bot, check this please

**Comment by yyongycy** [comment]
change for uri comparing

**Comment by yyongycy** [comment]
alluxio-bot, check this please

File: underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java [hunk 1]
```diff
@@ -763,13 +763,15 @@ private boolean delete(String path, boolean recursive) throws IOException {
...
     try {
-      files = hdfs.listStatus(new Path(path));
+      files = hdfs.listStatus(thePath);
     } catch (FileNotFoundException e) {
       return null;
     }
     // Check if path is a file
-    if (files != null && files.length == 1 && files[0].getPath().toString().equals(path)) {
+    if (files != null && files.length == 1 
+      && files[0].getPath().toUri().getPath().equals(thePath.toUri().getPath()) {
```
Comment (apc999) [inline underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java:774]: indentation and a missing `)`
```suggestion
        && files[0].getPath().toUri().getPath().equals(thePath.toUri().getPath())） {
```

**Comment by apc999** [comment]
@yyongycy thanks for the fix. I left a comment to pass the build and style check

**Comment by yyongycy** [comment]
> @yyongycy thanks for the fix. I left a comment to pass the build and style check

fixed already.

**Comment by yyongycy** [comment]
alluxio-bot, check this please

**Review by yyongycy — COMMENTED** [review]
modified accordingly.

**Review by apc999 — APPROVED** [review]

File: underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java [hunk 2]
```diff
@@ -763,13 +763,15 @@ private boolean delete(String path, boolean recursive) throws IOException {
...
+    Path thePath = new Path(path);
     try {
-      files = hdfs.listStatus(new Path(path));
+      files = hdfs.listStatus(thePath);
     } catch (FileNotFoundException e) {
       return null;
     }
     // Check if path is a file
-    if (files != null && files.length == 1 && files[0].getPath().toString().equals(path)) {
+    if (files != null && files.length == 1
```
Comment (apc999) [inline underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java:773]: please remove trailing whitespace
```suggestion
    if (files != null && files.length == 1
```

**Comment by LuQQiu** [comment]
@yyongycy you can run `mvn checkstyle:checkstyle` locally to make sure you don't break the code style check.

File: underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java (hunk 2 above)
Comment (yyongycy) [reply underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java:773]: I ran the stylecheck locally, and the extra space does cause probllem. Modifying it.

**Comment by yyongycy** [comment]
> @yyongycy you can run `mvn checkstyle:checkstyle` locally to make sure you don't break the code style check.

Yes, this is much easier to debug.

**Comment by yyongycy** [comment]
two test failure, they are not likely related to my change?

============
Error: 8.803 [ERROR] Tests run: 10, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 628.328 s <<< FAILURE! - in alluxio.server.ft.journal.JournalBackupIntegrationTest
Error: 8.809 [ERROR] alluxio.server.ft.journal.JournalBackupIntegrationTest.backupRestoreEmbedded  Time elapsed: 166.929 s  <<< ERROR!



Error: 3.077 [ERROR] Errors: 
Error: 3.077 [ERROR] alluxio.server.ft.journal.JournalBackupIntegrationTest.backupRestoreEmbedded
Error: 3.078 [ERROR]   Run 1: JournalBackupIntegrationTest.backupRestoreEmbedded:139->backupRestoreTest:414 » Unavailable
Error: 3.079 [ERROR]   Run 2: JournalBackupIntegrationTest.backupRestoreEmbedded:139->backupRestoreTest:414 » Unavailable

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Comment by apc999** [comment]
> two test failure, they are not likely related to my change?
> 
> ============
> Error: 8.803 [ERROR] Tests run: 10, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 628.328 s <<< FAILURE! - in alluxio.server.ft.journal.JournalBackupIntegrationTest
> Error: 8.809 [ERROR] alluxio.server.ft.journal.JournalBackupIntegrationTest.backupRestoreEmbedded Time elapsed: 166.929 s <<< ERROR!
> 
> Error: 3.077 [ERROR] Errors:
> Error: 3.077 [ERROR] alluxio.server.ft.journal.JournalBackupIntegrationTest.backupRestoreEmbedded
> Error: 3.078 [ERROR] Run 1: JournalBackupIntegrationTest.backupRestoreEmbedded:139->backupRestoreTest:414 » Unavailable
> Error: 3.079 [ERROR] Run 2: JournalBackupIntegrationTest.backupRestoreEmbedded:139->backupRestoreTest:414 » Unavailable

yeah, they are flaky

**Pull request merged**